app/src/main/java/mozilla/lockbox/presenter/EditItemPresenter.kt

```diff
@@ -138,17 +133,18 @@ class EditItemPresenter(
         newUsername: CharSequence? = null,
         newPassword: CharSequence? = null
     ) {
-        try {
+        credentials?.let { cred ->
             credentials = ServerPassword(
-                id = credentials?.id.orEmpty(),
-                hostname = newHostname?.toString() ?: credentials?.hostname.orEmpty(),
-                username = newUsername?.toString() ?: credentials?.username,
-                password = newPassword?.toString() ?: credentials?.password.orEmpty(),
-                httpRealm = credentials?.httpRealm,
-                formSubmitURL = credentials?.formSubmitURL
+                id = cred.id,
+                hostname = newHostname?.toString() ?: cred.hostname,
+                username = newUsername?.toString() ?: cred.username,
+                password = newPassword?.toString() ?: cred.password,
+                httpRealm = cred.httpRealm,
+                formSubmitURL = cred.formSubmitURL
             )
-        } catch (exception: NullPointerException) {
-            pushError(exception, "Error editing credential with id ${credentials?.id}")
-        }
+        } ?: pushError(
+            NullPointerException("Credentials are null"),
+            "Error editing credential with id ${credentials?.id}"
+        )
```

Comment:
:sparkles: This code is much much better; not least because this works as intended. :sparkles:

Reply:
🚀